app/models/shopping_list.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+# ShoppingList
+class ShoppingList < ApplicationRecord
+  belongs_to :item
+
+  # if player inventory has item, is in current level, and quantity matches
+  # shopping list, then item has been obtained
+  def item_obtained(player)
+    Inventory.where(character: player).each do |inventory_item|
+      return true if inventory_item.item == item && inventory_item.quantity >= quantity
+    end
```

Comment:
let's use the any function? probably more suitable for this

https://www.educative.io/answers/what-is-any-used-for-in-an-array-in-ruby